### ISSUE TITLE
dsl: JsonSchema.__eq__ ignores whitespace_pattern, causing stale cache hits

### ISSUE THREAD

**Issue by harsh4vardhan** [outlines-dev/outlines#1981]
`JsonSchema.__eq__` (lines 447–455) compares only the parsed schema JSON, never `whitespace_pattern`:

```python
def __eq__(self, other):
    if not isinstance(other, JsonSchema):
        return False
    try:
        self_dict = json.loads(self.schema)
        other_dict = json.loads(other.schema)
        return self_dict == other_dict   # whitespace_pattern not compared
    except json.JSONDecodeError:
        return self.schema == other.schema
```

`whitespace_pattern` materially changes the regex produced by `to_regex`. Any code that caches or deduplicates logits processors keyed on `JsonSchema` equality (e.g. via `@cache`, a `dict`, or a set) will serve a processor built for the first `whitespace_pattern` when the same schema is reused with a different pattern — silently producing incorrectly constrained output.

**Minimum example:**

```python
from outlines.types.dsl import JsonSchema

schema = '{"type": "string"}'
j1 = JsonSchema(schema, whitespace_pattern=r"[ ]?")
j2 = JsonSchema(schema, whitespace_pattern=r"[\n\t ]*")

assert j1 == j2   # True — wrong; patterns differ
```

`CFG.__eq__` correctly compares its single attribute. `JsonSchema.__eq__` should do the same for `whitespace_pattern`.

**Fix:** add `and self.whitespace_pattern == other.whitespace_pattern` to the equality check.
